hs_core/hydroshare/hs_bagit.py

```diff
@@ -41,217 +50,104 @@ def delete_files_and_bag(resource):
     for bag in resource.bags.all():
         bag.delete()
 
+def create_bag(resource):
 
-def create_bag_files(resource):
-    """
-    create and update files needed by bagit operation that is conducted on iRODS server;
-    no bagit operation is performed, only files that will be included in the bag are created
-    or updated.
+    checksums = ['md5', 'sha256']
 
-    Parameters:
-    :param resource: A resource whose files will be created or updated to be included in the
-    resource bag.
-    :return: istorage, an IrodsStorage object that will be used by subsequent operation to
-    create a bag on demand as needed.
-    """
-    from hs_core.hydroshare.utils import current_site_url, get_file_mime_type
+    # generate remote-file-mainfest for fetch.txt
+    # generate metatdata json for bag-info.txt
+    remote_file_manifest_json = get_remote_file_manifest(resource)
+    metadata_json = get_metadata_json(resource)
 
-    istorage = resource.get_irods_storage()
+    tmpdir = os.path.join(settings.TEMP_FILE_DIR, uuid4().hex, resource.short_id)
+    os.makedirs(tmpdir)
 
-    # the temp_path is a temporary holding path to make the files available to iRODS
-    # we have to make temp_path unique even for the same resource with same update time
-    # to accommodate asynchronous multiple file move operations for the same resource
+    # create the remote manifest and metadata files
+    remote_manifest_file = os.path.join(tmpdir, 'remote-file-manifest.json')
+    with open(remote_manifest_file, 'w') as out:
+        out.write(remote_file_manifest_json)
```

Comment:
Make sure you clean up temp dir every time after it is created without leaving temp directory behind.